### PR TITLE
feat: add support for storing JSON schema

### PR DESCRIPTION
**Pull request by stakach**

### CODE REVIEW

**Comment by caspiano** [comment]
What about `ControlSystems`?
Also might be better to have this as a separate model, similar to settings. This will simplify the fetching and merging of JSON schemas for the frontend.

**Comment by stakach** [comment]
shouldn't be required for systems as it's basically just the sum of all logic module settings
I probably shouldn't even have it in zones either - it should actually be in metadata

**Comment by caspiano** [comment]
How does fetching and setting schemas for metadata work? Is it stored in the metadata?

File: src/placeos-models/zone.cr [hunk 1]
```diff
@@ -15,6 +16,7 @@ module PlaceOS::Model
     attribute name : String, es_subfield: "keyword"
     attribute description : String = ""
     attribute tags : Set(String) = ->{ Set(String).new }
+    attribute json_schema : JSON::Any = JSON::Any.new("{}"), converter: JSON::Any::StringConverter
```
Comment (caspiano) [inline src/placeos-models/zone.cr]: Saves a parse.
```suggestion
    attribute json_schema : JSON::Any = JSON::Any.new({} of JSON::Any => JSON::Any), converter: JSON::Any::StringConverter
```

**Comment by stakach** [comment]
> How does fetching and setting schemas for metadata work? Is it stored in the metadata?

yeah stored in metadata, basically just to ensure the frontend configuration is correct when someone edits it

**Comment by MrYuion** [comment]
@stakach, can we just store a string in the metadata and allow for storing schemas in a seperate table to allow reuse across metadata?

**Comment by stakach** [comment]
so a schema table and then allow metadata to implement a schema?

File: src/placeos-models/metadata.cr [hunk 2]
```diff
@@ -18,7 +18,6 @@ module PlaceOS::Model
     attribute description : String = ""
     attribute details : JSON::Any, converter: JSON::Any::StringConverter
     attribute editors : Set(String) = ->{ Set(String).new }
-    attribute schema : String = ""
 
     attribute parent_id : String, es_type: "keyword"
```
Comment (caspiano) [inline src/placeos-models/metadata.cr:22]:
```suggestion
    attribute parent_id : String, es_type: "keyword"
    attribute schema_id : String?, es_type: "keyword"
```

**Pull request merged**